### ISSUE TITLE
Impossible to Load file / Save file on Firefox x64

### ISSUE THREAD

**Issue by kigennek** [OXOIndustries/Trials-in-Tainted-Space#560]
I'm stuck on "Selected a file to load". No popup.

TiTS_0.6.28

**Comment by Gedan** [comment]
Gonna have to give us your file there I think. I just tested the latest release for myself in Firefox and it's loading from my test file perfectly fine.

**Comment by Gedan** [comment]
Actually no, I just fired up the 64-bit version for myself (just to make sure) and it's not working :|

**Comment by Gedan** [comment]
After some digging about, I'm fairly confident to place the blame *somewhere* in Firefox. The loading code is working fine across the standalone player, Chrome and FF 32-bit. The only browser that seems to be having an issue is FF 64-bit.

**Comment by Andre-Maartens** [comment]
mine is working
weird

**Comment by dragontamer8740** [comment]
Mine works too. Linux with 64 bit windows flash plugin in wine-pipelight. Flash player version 20.0.0.267.

**Comment by kigennek** [comment]
It still doesn't work for me on Windows 10 x64 and Flash 20.0.0.286. Could be a Windows only problem. Official Firefox 64-bit for Windows is new (2 months old maybe?).

**Comment by Gedan** [comment]
@dragontamer8740 No, when I went and looked to test this all for myself, I grabbed the installer straight from the FF site. No indication that it was ~new and fancy~ or anything. Once I did some digging around, I settled fairly quickly on this being down to a quirk of 64-Bit Windows FF having a few rough edges.

I suppose I should go properly log the issue, but I haven't had opportunity to actually throw together a simple repro-case project that I can give to people in a professional capacity...

**Comment by dragontamer8740** [comment]
@Gedan oh, okay. Weird that wine works better than actual windows for it.

Anyway, yeah, the bug reporting issue on porn games is problematic (this isn't my main github account of course, I assume the same is true for just about everyone else here.)

I wonder if it worked properly in the old waterfox builds?

**Comment by mirh** [comment]
If you check Windows System32/SysWOW64 folders, you can see there's a little difference. 
In the first there's just FlashUtil64.exe and NPSWF64.dll, while in the second -aside of the respective 32 bit files- there's also FlashPlayerPlugin.exe. Which is what handles the whole thing in the 32 bit firefox version (64 bit one just manages everything in plugin-container)

If you try the online version, you can even see the save dialog displaying (and they both seem to use comdlg32.dll). Then I loose every reference in ProcMon. 
TL;DR.. I wouldn't know if it's adobe's fault, or firefox not expecting a "neater plugin model". 

Perhaps somebody could write a testcase of the `file.save` function?

**Comment by ds6** [comment]
I'll test this on nightly x64 later to see if it will be fixed in an upcoming mainline release.

**Comment by mirh** [comment]
[99% it isn't](https://bugzilla.mozilla.org/show_bug.cgi?id=1284897).

**Comment by ds6** [comment]
I was going to check bugzilla to see if it was reported, thanks for saving me the trouble.

flash does some *weird shit* when I try to load or save a file; aside from the window not opening, the url bar no longer responds to input, but still allows me to change it (as in, I get no suggestions when typing anything but I still have full control over what I can put in there), and the search field doesn't do anything at all. additionally, TiTS just completely *disappears* for no real reason, leaving me with a blank window and a strangely hanging FF.

this seems to happen with e10s enabled as well, though the UI isn't unresponsive anymore. I also don't get any prompts about the tab being unresponsive.

**Comment by dragontamer8740** [comment]
Looks to me like this can be fixed by setting 'dom.ipc.plugins.sandbox-level.flash' to 1 in `about:config`.
